libs/data-access/categories/src/lib/saga.ts

```diff
@@ -0,0 +1,108 @@
+import { call, put, takeEvery } from 'redux-saga/effects';
+import { ActionTypes } from './constants';
+import {
+  getAllSuccess,
+  getAllFailed,
+  createSuccess,
+  createFailed,
+  updateSuccess,
+  updateFailed,
+  removeSuccess,
+  removeFailed,
+  getByIdSuccess,
+  getByIdFailed,
+} from './actions';
+import { post, get } from '@ppm/data-access/http-requests';
+import { PrivateRoutesPath } from '@ppm/common/main';
+
+export function* createCategory(actions) {
+  try {
+    const path = `/api/${PrivateRoutesPath.CATEGORIES}`;
+    const data = actions.payload;
+    yield call(post, path, data);
+    yield put(
+      createSuccess({
+        loading: false,
+      })
+    );
+  } catch (error) {
+    yield put(createFailed());
+  }
+}
+
+export function* updateCategory(actions) {
+  try {
+    const data = actions.payload;
+    const path = `/api/${PrivateRoutesPath.CATEGORIES}/update/${data.id}`;
+    yield call(post, path, data);
+    yield put(
+      updateSuccess({
+        loading: false,
+      })
+    );
+  } catch (error) {
+    yield put(updateFailed());
+  }
+}
+
+export function* removeCategory(actions) {
+  try {
+    const id = actions.payload;
+    const path = `/api/${PrivateRoutesPath.CATEGORIES}/delete/${id}`;
+    const result = yield call(post, path);
+    if (result) {
+      yield put(removeFailed());
```

Comment:
on this case if your call return something it make action failed and after that also makes removeSuccess action, because this if statement doesn't brake this function.